app/src/main/java/com/chess/clock/activities/ClockSoundManager.java

```diff
@@ -69,31 +70,44 @@ public boolean areSoundsEnabled() {
     }
 
     @Override
-    public void playSound(ClockSound sound) {
+    public void playSound(ClockSound sound, AudioManager manager) {
         if (!soundsEnabled) return;
-        if (!preparedSounds.contains(sound.name())) return;
+        int soundId = clockSoundId(sound);
+
+        if (soundId == NO_SOUND_ID) return; // sound failed to load
+        if (!preparedSoundsIds.contains(soundId)) return; // sound not prepared
+
+        float actualVolume = (float) manager.getStreamVolume(AudioManager.STREAM_MUSIC);
+        float maxVolume = (float) manager.getStreamMaxVolume(AudioManager.STREAM_MUSIC);
+        float soundVolume = actualVolume / maxVolume;
```

Comment:
I hope `getStreamMaxVolume`  cannot return 0 :D

Reply:
Me too :D

Reply:
It shouldn't but will see.I think this should show crashes pretty fast and we'll update this ASAP if any that case happen.